### DISCUSSION TITLE
How should I convert {{$guid}} from postman

### DISCUSSION THREAD

**Discussion by natt-han** [usebruno/bruno#1073] [Q&A] [answered]
Hello,

I'm migrating my collection from postman which using {{$guid}} in request boby.
how should I do that in bruno.

Thanks.

> [!NOTE]
> Bruno now supports random variables. You can use `{{$guid}}`

**Comment by martinsefcik** [comment, marked as answer]
For example you can use this code in pre-request script
```js
bru.setVar('uuid', require("uuid").v4());
```
to set collection variable `uuid` with uuid value.

then you can use it e.g. in request body or url like this:
```
{{uuid}}
```

Currently I guess there is no simpler way to achieve that, but as far I know there are plans to add this predefined variables like in Postman.

**Comment by helloanoop** [comment]
@natt-han 

We will soon support `$guid` in the future.

As @martinsefcik wrote https://github.com/usebruno/bruno/discussions/1073#discussioncomment-7690126, you can do it at a request level.

If you want to have this written one time and have it applied to any requests, please update the pre request script in the collection settings.
You can open the collection settings by clicking on the gear icon in the top right side.

![image](https://github.com/usebruno/bruno/assets/9350733/43956c0c-2e68-4604-ba84-d97c1626cce1)

**Reply by nhomble** [reply]
> We will soon support `$guid` in the future.

afaict this isn't available yet, I am wondering if it's still recommended to add these helpers to my collections. I'd be happy to contribute as well if this feature is still under consideration.

**Reply by Its-treason** [reply]
> > We will soon support `$guid` in the future.
> 
> afaict this isn't available yet, I am wondering if it's still recommended to add these helpers to my collections. I'd be happy to contribute as well if this feature is still under consideration.

I don't see this added any time soon. Use the helper for now.

**Reply by so-jelly** [reply]
@Its-treason use which helper?

**Reply by Its-treason** [reply]
> @Its-treason use which helper?

The script Anoop posted above.

**Reply by Kars-de-Jong-TenneT** [reply]
Unfortunately, this isn't really an alternative if you need more than one GUID in your request. Using `{{uuid}}` multiple times within the same request will give the same uuid for every use.

It would help a great deal if something like this was possible:
```javascript
bru.setVar('uuid', () => require("uuid").v4());
```

**Reply by RaghavBhat02** [reply]
Please add support for this soon. It'd be a great help for me.

**Reply by Its-treason** [reply]
> Please add support for this soon. It'd be a great help for me.

It's already available in the latest version see: https://docs.usebruno.com/testing/script/dynamic-variables

**Comment by anusree-bruno** [comment]
In the recent versions we have added support for random variables and you can directly use {{$guid}}.
